src/main/drivers/pwm_output_dshot.c

```diff
@@ -116,7 +116,7 @@ void pwmDshotMotorHardwareConfig(const timerHardware_t *timerHardware, uint8_t m
         RCC_ClockCmd(timerRCC(timer), ENABLE);
         TIM_Cmd(timer, DISABLE);
 
-        TIM_TimeBaseStructure.TIM_Prescaler = (uint16_t)((timerClock(timer) / getDshotHz(pwmProtocolType)) - 1);
+        TIM_TimeBaseStructure.TIM_Prescaler = (uint16_t)(lrintf((float) timerClock(timer) / getDshotHz(pwmProtocolType) + 0.01f) - 1);
```

Comment:
IMO the value should be always rounded up to guarantee that DSHOT timing is on safe side.
```
int hz = getDshotHz(pwmProtocolType);
TIM_TimeBaseStructure.TIM_Prescaler = (uint16_t)(timerClock(timer) + hz - 1) / hz;